include/xsimd/types/xsimd_batch.hpp

```diff
@@ -22,6 +22,17 @@
 namespace xsimd
 {
 
+    namespace details
+    {
+        template <class B>
+        void check_batch_init(B const&, std::initializer_list<typename B::value_type> data)
+        {
+            (void)data;
+            assert(data.size() == B::size && "consistent initialization");
```

Comment:
It should be possible to do a static assert on `B::size` since it's constexpr.

Reply:
It's not possible because `data.size()` is not constexpr in C++11. It is in C++14 but anyway data here is not in a `constexpr` context so no :-/